pkg/operator/reconciler.go

```diff
@@ -0,0 +1,402 @@
+			return reconcile.Result{}, err
+		}
+	*/
+	return reconcile.Result{}, nil
+}
+
+func (r *Reconciler) syncDaemonSet(ctx context.Context, instance *configv1.ClusterOperator) error {
+	dsMf := ntomf.TunedDaemonSet()
+	controllerutil.SetControllerReference(instance, dsMf, r.Scheme)
+	ds := &appsv1.DaemonSet{}
+	key := types.NamespacedName{
+		Name:      dsMf.Name,
+		Namespace: r.Namespace,
+	}
+	err := r.Client.Get(ctx, key, ds)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.V(2).Infof("syncDaemonSet(): DaemonSet %s not found, creating one", dsMf.Name)
+			if err := r.Client.Create(ctx, dsMf); err != nil {
+				klog.Errorf("failed to create DaemonSet %s: %v", dsMf.Name, err)
+				return err
+			}
+			// DaemonSet created successfully
+			return nil
+		}
+		klog.Errorf("failed to get DaemonSet %s: %v", dsMf.Name, err)
+		return err
+	}
+
+	operatorReleaseVersion := os.Getenv("RELEASE_VERSION")
+	operandReleaseVersion := ""
+
+	for _, e := range ds.Spec.Template.Spec.Containers[0].Env {
+		if e.Name == "RELEASE_VERSION" {
+			operandReleaseVersion = e.Value
+			break
+		}
+	}
+
+	ds = ds.DeepCopy()
+	ds.Spec = dsMf.Spec
+
+	if operatorReleaseVersion != operandReleaseVersion {
+		// Update the DaemonSet
+		klog.V(2).Infof("syncDaemonSet(): operatorReleaseVersion (%s) != operandReleaseVersion (%s), updating", operatorReleaseVersion, operandReleaseVersion)
+		if err := r.Client.Update(ctx, ds); err != nil {
+			klog.Errorf("failed to update DaemonSet %s: %v", ds.Name, err)
+			return err
+		}
+		// DaemonSet updated successfully
+		return nil
+	}
+	// DaemonSet comparison is non-trivial and expensive
+	klog.V(2).Infof("syncDaemonSet(): found DaemonSet %s [%s], not changing it", ds.Name, operatorReleaseVersion)
+	return nil
+}
+
+func (r *Reconciler) syncDefaultTuned(ctx context.Context, instance *configv1.ClusterOperator) error {
+	crMf := ntomf.TunedCustomResource()
+	controllerutil.SetControllerReference(instance, crMf, r.Scheme)
+	cr := &tunedv1.Tuned{}
+	key := types.NamespacedName{
+		Name:      crMf.Name,
+		Namespace: r.Namespace,
+	}
+	err := r.Client.Get(ctx, key, cr)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.V(2).Infof("syncDefaultTuned(): Default Tuned %s not found, creating one", crMf.Name)
+			if err := r.Client.Create(ctx, crMf); err != nil {
+				klog.Errorf("failed to create Default Tuned %s: %v", crMf.Name, err)
+				return err
+			}
+			// Tuned resource created successfully
+			return nil
+		}
+		klog.Errorf("failed to get Default Tuned %s: %v", crMf.Name, err)
+		return err
+	}
+	// Tuned resource found, check whether we need to update it
+	if reflect.DeepEqual(crMf.Spec.Profile, cr.Spec.Profile) &&
+		reflect.DeepEqual(crMf.Spec.Recommend, cr.Spec.Recommend) {
+		klog.V(2).Infof("syncTunedDefault(): Tuned %s doesn't need updating", crMf.Name)
+		return nil
+	}
+	cr = cr.DeepCopy() // never update the objects from cache
```

Comment:
Is this still needed or does the controller-runtime client handle this by not returning objects from cache?

Reply:
We still need this for handling mutated objects for updates (@cynepco3hahue keep me honest here)
Maybe we should change the name of the recipient to  mutatedCr / mutatedDefaultTuned